components/brave_rewards/common/rewards_flags.cc

```diff
@@ -83,18 +84,25 @@ RewardsFlags RewardsFlags::Parse(const std::string& input) {
   return flags;
 }
 
-const RewardsFlags& RewardsFlags::ForCurrentProcess() {
-  static RewardsFlags parsed_flags = [] {
-    std::string input;
+}  // namespace
 
-    const auto* command_line = base::CommandLine::ForCurrentProcess();
-    if (command_line->HasSwitch(kSwitchName)) {
-      input = command_line->GetSwitchValueASCII(kSwitchName);
-    }
+void RewardsFlags::ForceParsingForTesting() {
+  g_force_parse_for_testing = true;
+}
+
+RewardsFlags RewardsFlags::ForCurrentProcess() {
```

Comment:
BTW, we can do something like this without changing of return value:

```
const RewardsFlags& RewardsFlags::ForCurrentProcess() {
  static absl::optional<RewardsFlags> parsed_flags;

  if (parsed_flags && !g_force_parse_for_testing) {
    return *parsed_flags;
  }

  std::string input;

  const auto* command_line = base::CommandLine::ForCurrentProcess();
  if (command_line->HasSwitch(kSwitchName)) {
    input = command_line->GetSwitchValueASCII(kSwitchName);
  }

  parsed_flags = Parse(input);

  return *parsed_flags;
}

```

Reply:
I like this suggestion.

Reply:
Fixed